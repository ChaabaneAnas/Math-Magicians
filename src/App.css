body {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
}

#root {
  width: 60%;
  max-width: 500px;
  height: 40%;
  text-align: center;
}

.app-container {
  height: 100%;
}

.display {
  text-align: right;
  color: white;
  padding: 27px;
  background-color: grey;
}

.second-row {
  display: flex;
  height: 100%;
}

.keyboard {
  flex: 3;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: repeat(5, 1fr);
  align-content: stretch;
  gap: 1px;
}

.keyboard div {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: lightgrey;
}

.operations {
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: stretch;
  gap: 1px;
}

.operations div {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: orange;
}

.keyboard div:hover {
  background-color: rgba(211, 211, 211, 0.541);
}

.Zero {
  grid-column: span 2;
}

.operations div:hover {
  background-color: rgba(255, 166, 0, 0.493);
}
