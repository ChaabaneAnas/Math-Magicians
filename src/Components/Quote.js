const Quote = () => (
  <p className="Quote">
    mathematics is not about numbers equations computations or algorithms its about understanding.
    ~~ William Paul thurston
  </p>
);

export default Quote;
