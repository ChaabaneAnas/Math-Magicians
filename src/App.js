/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import { Display, Keyboard, Operations } from './Components/calculator';

class App extends React.Component {
  render() {
    return (
      <div className="app-container">
        <Display />
        <Operations />
        <Keyboard />
      </div>
    );
  }
}
