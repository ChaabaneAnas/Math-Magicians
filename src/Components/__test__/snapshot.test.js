import React from 'react';
import Header from '../header';
import Calculator from '../calculator';
import App from '../../App';
import Quote from '../Quote';
import { Display } from '../calculatorComponent';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import Home from '../home';

afterEach(cleanup);

// Testing Calculator Component
it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Calculator display should equal to 0', () => {
  const { getByText } = render(<Calculator />);
  expect(getByText(/0/i)).toMatchSnapshot();
});

test('Testing Events: Calculator display should equal to 85', () => {
  const { getByText } = render(<Calculator />);
  fireEvent.click(getByText(/8/i, { selector: 'button' }));
  fireEvent.click(getByText(/5/i, { selector: 'button' }));
  const display = screen.getByText(/85/i, { selector: '.display' });
  expect(parseInt(display.textContent)).toBe(85);
});

test('Testing Events: Calculator display should equal to 50', () => {
  const { getByText } = render(<Calculator />);
  fireEvent.click(getByText(/2/i, { selector: 'button' }));
  fireEvent.click(getByText(/x/i, { selector: 'button' }));
  fireEvent.click(getByText(/5/i, { selector: 'button' }));
  fireEvent.click(getByText(/x/i, { selector: 'button' }));
  fireEvent.click(getByText(/5/i, { selector: 'button' }));
  fireEvent.click(getByText(/=/i, { selector: 'button' }));
  const display = screen.getByText(/50/i, { selector: '.display' });
  expect(parseInt(display.textContent)).toBe(50);
});
// testing Home Component

test('Testing Links: Click on Home should render Home page', () => {
  render(<Home />);
  const display = screen.getByText(/Welcome to our page/i);
  expect(display).toMatchSnapshot();
});

//Testing Quote Component

test('Testing Links: Click on Home should render Home page', () => {
  render(<Quote />);
  const display = screen.getByText(/Mathematics/i);
  expect(display).toMatchSnapshot();
});
