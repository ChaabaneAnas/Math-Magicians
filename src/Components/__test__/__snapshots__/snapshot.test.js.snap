// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator display should equal to 0 1`] = `
<button
  class="Zero"
  type="button"
>
  0
</button>
`;

exports[`Testing Links: Click on Home should render Home page 1`] = `
<h2>
  Welcome to our page
</h2>
`;

exports[`Testing Links: Click on Home should render Home page 2`] = `
<p
  class="Quote"
>
  mathematics is not about numbers equations computations or algorithms its about understanding. ~~ William Paul thurston
</p>
`;

exports[`renders correctly 1`] = `
<div
  className="main"
>
  <h2>
     Lets Do some Math 
  </h2>
  <div
    className="app-container"
  >
    <div
      className="display"
    />
    <div
      className="second-row"
    >
      <div
        className="keyboard"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="Zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
      <div
        className="operations"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          /
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
